docker-entrypoint-funcs.sh

```diff
@@ -135,87 +135,133 @@ edbdocker_run_entrypoint_parts() {
+  if [ -n "${EDGEDB_SERVER_PASSWORD:-}" ] &&
+     [ -n "${EDGEDB_SERVER_PASSWORD_HASH:-}" ]; then
+    edbdocker_die "ERROR: EDGEDB_SERVER_PASSWORD and EDGEDB_SERVER_PASSWORD_PASH are mutually exclusive, but both are set"
   fi
 
-  if [ -n "${EDGEDB_BOOTSTRAP_SCRIPT_FILE:-}" ] && [ -n "${EDGEDB_BOOTSTRAP_COMMAND:-}" ]; then
-    edbdocker_die "ERROR: EDGEDB_BOOTSTRAP_SCRIPT_FILE and EDGEDB_BOOTSTRAP_COMMAND are mutually exclusive, but both are set"
+  if [ -n "${EDGEDB_SERVER_BOOTSTRAP_SCRIPT_FILE:-}" ] &&
+     [ -n "${EDGEDB_SERVER_BOOTSTRAP_COMMAND:-}" ]; then
+    edbdocker_die "ERROR: EDGEDB_SERVER_BOOTSTRAP_SCRIPT_FILE and EDGEDB_SERVER_BOOTSTRAP_COMMAND are mutually exclusive, but both are set"
   fi
 
-  export EDGEDB_RUNSTATE_DIR="${EDGEDB_RUNSTATE_DIR:-/run/edgedb}"
+  export EDGEDB_SERVER_RUNSTATE_DIR="${EDGEDB_SERVER_RUNSTATE_DIR:-/run/edgedb}"
 }
 
 
 # Resolve the value of the specified variable.
 # Usage: edbdocker_lookup_env_var VARNAME [default]
 # The function looks for $VARNAME in the environment block directly,
 # and also tries to read the value from ${VARNAME}_FILE, if set.
-# For example, `edbdocker_lookup_env_var EDGEDB_PASSWORD foo` would
-# look for $EDGEDB_PASSWORD, the file specified by $EDGEDB_PASSWORD_FILE,
-# and if neither is set, default to 'foo'.
+# For example, `edbdocker_lookup_env_var EDGEDB_SERVER_PASSWORD foo` would
+# look for $EDGEDB_SERVER_PASSWORD, the file specified by
+# $EDGEDB_SERVER_PASSWORD_FILE, and if neither is set, default to 'foo'.
 edbdocker_lookup_env_var() {
   local var="$1"
   local file_var="${var}_FILE"
+  local old_var="${var/EDGEDB_SERVER_/EDGEDB_}"
+  local old_file_var="${old_var}_FILE"
   local deflt="${2:-}"
   local val="$deflt"
   local var_val="${!var:-}"
   local file_var_val="${!file_var:-}"
+  local old_var_val="${!old_var:-}"
+  local old_file_var_val="${!old_file_var:-}"
 
-  if [ -n "${var_val}" ] && [ -n "${file_var_val}" ]; then
-    edbdocker_die \
-      "ERROR: ${var} and ${file_var} are exclusive, but both are set."
+  if [ -n "${old_var_val}" ]; then
+    msg=(
+      "=============================================================== "
+      "WARNING: ${old_var} is deprecated use ${var} instead. "
+      "=============================================================== "
+    )
+    edbdocker_log "${msg[@]}"
+  fi
+
+  if [ -n "${old_file_var_val}" ]; then
+    msg=(
+      "=============================================================== "
+      "WARNING: ${old_file_var} is deprecated use ${file_var} instead. "
+      "=============================================================== "
+    )
+    edbdocker_log "${msg[@]}"
+  fi
+
+  local -A exclusive_pairs=(
+    ["${var}"]="${file_var}"
+    ["${var}"]="${old_var}"
+    ["${var}"]="${old_file_var}"
+    ["${old_var}"]="${file_var}"
+    ["${old_var}"]="${old_file_var}"
+    ["${file_var}"]="${old_file_var}"
```

Comment:
This implementation will not check for usage of mutually exclusive variable pairs in any particular order. Is that acceptable?